lib/galaxy/tool_util/parser/interface.py

```diff
@@ -290,6 +290,25 @@ def get_bool(self, key, default):
     def parse_label(self):
         return self.get("label")
 
+    def parse_name(self):
+        """Return name of an input source
+        returns the name or if absent the argument property
+        in the later case leading dashes are stripped and
+        internal dashes are replaced by underscore
+        """
+        return self._parse_name(self.get('name'), self.get('argument'))
+
+    @staticmethod
+    def _parse_name(name, argument):
```

Comment:
@bernt-matthias  Now that I see it implemented, I agree with you that a function in `lib/galaxy/tool_util/parser/util.py` would look better (when imported in the linter) than this static method, but I still like having the `parse_name()` method in this class. Would you agree to move this and keep the one above?

Sorry for going back and forth on this.

Reply:
No worries. Coding is an iterative process :) 

Just trying to wrap my head around the failing tests ... I assumed that this `input_source` actually is an `InputSource`

https://github.com/galaxyproject/galaxy/blob/0f2ab9a08da99133541ea0b5f4b51467ef390bd9/lib/galaxy/tools/parameters/basic.py#L255

seems not to be the case.

Reply:
Riddle after riddle. Any idea why these imports are not working?

Reply:
There are now circular imports between lib/galaxy/tool_util/parser/util.py and lib/galaxy/tool_util/parser/interface.py

Reply:
Good question, I'm not sure I care between the bad choices. Refactor the ToolStdio classes into util or revert to using a staticmethod without the import, or move the utility into the interface class. They are all more or less fine and not particularly great.

Reply:
I think you forgot to remove the static method from here.

Reply:
indeed. fixed